src/hotspot/share/gc/z/zArguments.cpp

```diff
@@ -72,6 +72,11 @@ void ZArguments::initialize() {
     vm_exit_during_initialization("The flag -XX:+UseZGC can not be combined with -XX:ConcGCThreads=0");
   }
 
+  // Only 2M large pages is supported
+  if (!FLAG_IS_DEFAULT(LargePageSizeInBytes) && LargePageSizeInBytes != 2 * M) {
+    vm_exit_during_initialization("Invalid -XX:LargePageSizeInBytes (only 2M large pages is supported)");
+  }
```

Comment:
To better handle the case where the default large page size is not supported I suggest we add something like:
```
if (LargePageSizeInBytes != 2 * M) {
  if (FLAG_IS_DEFAULT(LargePageSizeInBytes)) {
    vm_exit_during_initialization("Default large page size is not supported (only 2M large pages are supported)");
  } else {
    vm_exit_during_initialization("Invalid -XX:LargePageSizeInBytes (only 2M large pages are supported)");
  }
}
```
Probably good to include the default size in the above print.

Reply:
I talked to Stefan offline and we agreed to not do this change, since it's not quite the behavior we want.